### ISSUE TITLE
OBB vision/arm orientation

### ISSUE THREAD

**Issue by Pweder3** [chsbacon/FTC-INTO-THE-DEEP-7080#11]
OBB vision or Object Orient Bounding Boxes Vision is an object detection algorithm that uses a camera to detect the positional location of a specific object while aligning a rotated bounding box with the object. The rotation part is vital to the robot as the camera is supposed to help the arm automatically rotate to the correct position. This would aim to increase the accuracy and speed of our cycle times. 

### **Vison model Information**
1. We will probably use a [YOLO11-OBB pre-trained model ](https://docs.ultralytics.com/models/yolo11/#supported-tasks-and-modes).
2. This model needs around 50-75 training images (educated guess) to get what we want 

### **Potential problems**

- [ ] algorithm could be to slow and cause the robot controller to hang
    - This can be tested by using different [size models](https://docs.ultralytics.com/models/yolo11/#supported-tasks-and-modes) (trades off for accuracy)

| Model       | size (pixels) | mAPtest 50 | Speed CPU ONNX (ms) | Speed T4 TensorRT10 (ms) | params (M) | FLOPs (B) |
|-------------|---------------|------------|---------------------|--------------------------|------------|-----------|
| YOLO11n-obb | 1024          | 78.4       | 117.6 ± 0.8         | 4.4 ± 0.0                | 2.7        | 17.2      |
| YOLO11s-obb | 1024          | 79.5       | 219.4 ± 4.0         | 5.1 ± 0.0                | 9.7        | 57.5      |
| YOLO11m-obb | 1024          | 80.9       | 562.8 ± 2.9         | 10.1 ± 0.4               | 20.9       | 183.5     |
| YOLO11l-obb | 1024          | 81.0       | 712.5 ± 5.0         | 13.5 ± 0.6               | 26.2       | 232.0     |
| YOLO11x-obb | 1024          | 81.3       | 1408.6 ± 7.7        | 28.6 ± 1.0               | 58.8       | 520.2     |
> Performance table for reference

---
- [ ] There could be a problem converting the model to a Tensorflow Light file (.TFlite)
---
- [ ] The camera could be bad enough or grainy enough not to give accurate data
---
- [ ] Our FOV could be to small to gather meaning full

**Comment by Pweder3** [comment]
1. Have completed the docker containerization setup and tested the preliminary image collection. Im using a software called Label Studio to create my datasets as it allows me to easily label the specimen and export the data In the correct formatt.

2. might have to train the model through my own script and not through Label studio

3. found a [writeup ](https://docs.ultralytics.com/modes/export/#why-choose-yolo11s-export-mode) on exporting yolo models

**Comment by Pweder3** [comment]
Have collected images and plan to train the model before the next in-person meeting although transporting the model will be hard as it could be surprisingly large
